clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,912 @@
+	insertMatches := reSubMatchMap(insertRgx, cmd)
+	insertIntoMatches := reSubMatchMap(insertIntoRgx, cmd)
+	if insertIntoMatches != nil {
+		db = getIdentFromMatches(insertIntoMatches, "db")
+		rp = getIdentFromMatches(insertIntoMatches, "rp")
+		point = getIdentFromMatches(insertIntoMatches, "point")
+		if db != "" && rp == "" {
+			defaultRp, err := c.getDefaultRetentionPolicy(context.Background(), db)
+			if err != nil {
+				color.Red("Unable to get default retention policy for %q", db)
+				return
+			}
+			rp = defaultRp
+		}
+	} else if !strings.HasPrefix(strings.ToUpper(cmd), "INSERT INTO") && insertMatches != nil {
+		db = c.Database
+		rp = c.RetentionPolicy
+		point = getIdentFromMatches(insertMatches, "point")
+		if db == "" {
+			color.Red("Please run \"use <database>\" to run \"INSERT <point\"")
+			return
+		}
+	} else {
+		color.Red("Expected \"INSERT INTO <database>.<retention_policy> <point>\" OR \"INSERT <point>\".")
+		return
+	}
+
+	buf := bytes.Buffer{}
+	gzw := gzip.NewWriter(&buf)
+
+	_, err := gzw.Write([]byte(point))
+	gzw.Close()
+	if err != nil {
+		color.Red("Failed to gzip points")
+		return
+	}
+	bucketID, err := c.getBucketIdForDBRP(db, rp)
+	if err != nil {
+		color.Red("Unable to match DBRP to BucketID: %v", err)
+		return
+	}
+	writeReq := c.PostWrite(context.Background()).
+		Bucket(bucketID).
+		Precision(api.WritePrecision(c.Precision)).
+		ContentEncoding("gzip").
+		Body(buf.Bytes())
+	if c.OrgID != "" {
+		writeReq = writeReq.Org(c.OrgID)
+	} else if c.OrgName != "" {
+		writeReq = writeReq.Org(c.OrgName)
+	} else {
+		writeReq = writeReq.Org(c.ActiveConfig.Org)
+	}
+	if err := writeReq.Execute(); err != nil {
+		color.Red("ERR: %v", err)
+		if c.Database == "" {
+			fmt.Println("Note: error may be due to not setting a database or retention policy.")
+			fmt.Println(`Please set a database with the command "use <database>"`)
+			return
+		}
+	}
+}
+
+// Reverse search for a bucket from db & rp
+func (c Client) getBucketIdForDBRP(db string, rp string) (string, error) {
+	dbrpsReq := c.GetDBRPs(context.Background())
+	if c.OrgID != "" {
+		dbrpsReq = dbrpsReq.OrgID(c.OrgID)
+	}
+	if c.OrgName != "" {
+		dbrpsReq = dbrpsReq.Org(c.OrgName)
+	}
+	if c.OrgID == "" && c.OrgName == "" {
+		dbrpsReq = dbrpsReq.Org(c.ActiveConfig.Org)
+	}
+	dbrps, err := dbrpsReq.Execute()
+	if err != nil {
+		return "", err
+	}
+	bucketID := ""
+	for _, dbrp := range *dbrps.Content {
+		if dbrp.Database == db && dbrp.RetentionPolicy == rp {
+			bucketID = dbrp.BucketID
+			break
+		}
+	}
+	if bucketID == "" {
+		return "", fmt.Errorf("unable to find bucket ID for %q.%q", db, rp)
+	}
+	return bucketID, nil
+}
+
+type FormatType string
+
+var (
+	CsvFormat   FormatType = "csv"
+	JsonFormat  FormatType = "json"
+	TableFormat FormatType = "table"
+)
+
+func (c *Client) runAndShowQuery(query string) {
+	// ? Should this function have support to guide users trying to use deprecated InfluxQL queries
```

Comment:
I would say yes, could even link to the docs page showing which keywords are not allowed if that seems reasonable, that way its a one-time-mistake instead of one time per disallowed keyword